cardano-db-sync/src/Cardano/DbSync/Api.hs

```diff
@@ -199,52 +200,64 @@ getPrunes :: SyncEnv -> Bool
 getPrunes = do
   DB.pcmPruneTxOut . getPruneConsume
 
-fullInsertOptions :: Bool -> InsertOptions
-fullInsertOptions useLedger =
-  InsertOptions
-    { ioInOut = True
-    , ioUseLedger = useLedger
-    , ioShelley = True
-    , ioRewards = True
-    , ioMultiAssets = True
-    , ioMetadata = True
-    , ioKeepMetadataNames = Strict.Nothing
-    , ioPlutusExtra = True
-    , ioOffChainPoolData = True
-    , ioGov = True
+extractInsertOptions :: SyncPreConfig -> SyncInsertOptions
+extractInsertOptions cfg =
+  case pcInsertConfig cfg of
+    FullInsertOptions -> fullInsertOptions
+    OnlyUTxOInsertOptions -> onlyUTxOInsertOptions
+    OnlyGovInsertOptions -> onlyGovInsertOptions
+    DisableAllInsertOptions -> disableAllInsertOptions
+    SyncInsertConfig opts -> opts
+
+fullInsertOptions :: SyncInsertOptions
+fullInsertOptions =
+  SyncInsertOptions
+    { sioTxOut = TxOutEnable
+    , sioLedger = LedgerEnable
+    , sioShelley = ShelleyEnable
+    , sioRewards = RewardsConfig True
+    , sioMultiAsset = MultiAssetEnable
+    , sioMetadata = MetadataEnable
+    , sioPlutus = PlutusEnable
+    , sioGovernance = GovernanceConfig True
+    , sioOffchainPoolData = OffchainPoolDataConfig True
+    , sioJsonType = JsonTypeText
     }
 
-onlyUTxOInsertOptions :: InsertOptions
+onlyUTxOInsertOptions :: SyncInsertOptions
 onlyUTxOInsertOptions =
-  InsertOptions
-    { ioInOut = True
-    , ioUseLedger = False
-    , ioShelley = False
-    , ioRewards = False
-    , ioMultiAssets = True
-    , ioMetadata = False
-    , ioKeepMetadataNames = Strict.Nothing
-    , ioPlutusExtra = False
-    , ioOffChainPoolData = False
-    , ioGov = False
+  SyncInsertOptions
+    { sioTxOut = TxOutEnable
+    , sioLedger = LedgerIgnore
+    , sioShelley = ShelleyDisable
+    , sioRewards = RewardsConfig True
+    , sioMultiAsset = MultiAssetDisable
+    , sioMetadata = MetadataDisable
+    , sioPlutus = PlutusDisable
+    , sioGovernance = GovernanceConfig False
+    , sioOffchainPoolData = OffchainPoolDataConfig False
+    , sioJsonType = JsonTypeText
     }
 
-onlyGovInsertOptions :: Bool -> InsertOptions
-onlyGovInsertOptions useLedger = (disableAllInsertOptions useLedger) {ioGov = True}
-
-disableAllInsertOptions :: Bool -> InsertOptions
-disableAllInsertOptions useLedger =
-  InsertOptions
-    { ioInOut = False
-    , ioUseLedger = useLedger
-    , ioShelley = False
-    , ioRewards = False
-    , ioMultiAssets = False
-    , ioMetadata = False
-    , ioKeepMetadataNames = Strict.Nothing
-    , ioPlutusExtra = False
-    , ioOffChainPoolData = False
-    , ioGov = False
+onlyGovInsertOptions :: SyncInsertOptions
+onlyGovInsertOptions =
+  disableAllInsertOptions
+    { sioGovernance = GovernanceConfig True
+    }
+
+disableAllInsertOptions :: SyncInsertOptions
+disableAllInsertOptions =
+  SyncInsertOptions
+    { sioTxOut = TxOutDisable
+    , sioLedger = LedgerDisable
```

Comment:
For `onlyGovInsertOptions` the ledger should be enabled by default.
Ideally the ledger should remain an option even with preset insert configuration. For this pr it would be enough to change the only gov to have `LedgerEnable` by default.